doc: add meeting minutes (#31)
nodejs/performance: meetings/2022-11-28.md

# Node.js  Performance Team Meeting 2022-11-28

## Links

* **Recording**:  https://www.youtube.com/watch?v=f0yEem50VOI

> * **GitHub Issue**: $GITHUB_ISSUE$

Reviewer: ```suggestion
* **GitHub Issue**: https://github.com/nodejs/performance/issues/28
```

## Present

* Performance team: @nodejs/performance
* Yagiz Nizipli (@anonrig)
* Ulises Gascon (@ulisesgascon)
* Jean Burellier (@sheplu)
* Ethan Arrowood (@Ethan-Arrowood)
* Stephen Belanger (@qard)
* Juan Arboleda (@juanarbol)
* Marco Ippolito (@marco-ippolito)

> 

Reviewer: Stephen is missing from this list afaik

## Agenda

## Announcements

*Extracted from **performance-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/performance

* Add support for V8 fast call [#23](https://github.com/nodejs/performance/issues/23)
	* Multiple people are keen to help (Matteo, James, …)
	* stephen: V8 fast call API can be (very) limited but we need to investigated the gain / impact

> 	* Yagiz: find more information and update the PR

Reviewer: ```suggestion
	* Juan & Yagiz: find more information and come up with a PR
```

* Merging Github team with README [#19](https://github.com/nodejs/performance/issues/19)
	* Yagiz & Jean: discussion around GitHub team VS people listed in the README.md
* Buffer.toString vs TextDecoder [#18](https://github.com/nodejs/performance/issues/18)
	* Yagiz: still some issue for small string and we should investigate that
	* Jean: will try to keep a look at it
* fetch [#11](https://github.com/nodejs/performance/issues/11)
	* Ethan: streams are slow by default. Need to find a way to have faster stream (for Fetch but also a lot of other parts of node). Micro optimization is maybe not the best use case here (based on how it is used) and would be better to have one good optimization (based on time passed) 
    * spec => https://fetch.spec.whatwg.org/
    * Yagiz : Check turbofan for optimization
* 64 bit integers [#10](https://github.com/nodejs/performance/issues/10)
	* nothing new
* WHATWG Streams [#9](https://github.com/nodejs/performance/issues/9)
	* We will have a special discussion with multiple people around this topic. If you are interested you can add yourself on the issue 

> * Atomic condition variable [#5](https://github.com/nodejs/performance/issues/5)

Reviewer: ```suggestion
* Atomic condition variable [#5](https://github.com/nodejs/performance/issues/5)
  * removing agenda label due to missing benchmarks
```

* Buffer.indexOf [#4](https://github.com/nodejs/performance/issues/4)

> 	* to be closed

Reviewer: ```suggestion
	* Tony did a fantastic job, but unfortunately there isn't any room for improvement at the moment.
```

* Buffer.(read|write)(U)Int(LE|BE)(8|16|32) [#2](https://github.com/nodejs/performance/issues/2)

> 	* nothing new

Reviewer: ```suggestion
	* nothing new, moving to the next meetings agenda
```

## Q&A, Other

> * Yagiz: better version of array-join - could be nice to check  https://github.com/anonrig/fast-array-join

Reviewer: ```suggestion
* Yagiz: Open sourced @BridgeAR's faster version of array-join - could be nice to check  https://github.com/anonrig/fast-array-join
```
